hyperliquid/websocket_manager.py

```diff
@@ -46,19 +46,51 @@ def __init__(self, base_url):
         self.ws_ready = False
         self.queued_subscriptions: List[Tuple[Subscription, ActiveSubscription]] = []
         self.active_subscriptions: Dict[str, List[ActiveSubscription]] = defaultdict(list)
-        ws_url = "ws" + base_url[len("http") :] + "/ws"
-        self.ws = websocket.WebSocketApp(ws_url, on_message=self.on_message, on_open=self.on_open)
-        self.ping_sender = threading.Thread(target=self.send_ping)
+        self.base_url = base_url
+        self.ws = None
+        self.reconnect_interval = 5
+        self.should_reconnect = True
+        self.ping_sender = None
+
+    def start_ping_sender(self):
+        if self.ping_sender is None or not self.ping_sender.is_alive():
+            self.ping_sender = threading.Thread(target=self.send_ping)
+            self.ping_sender.start()
+
+    def connect(self):
+        ws_url = "ws" + self.base_url[len("http") :] + "/ws"
+        self.ws = websocket.WebSocketApp(
+            ws_url, on_message=self.on_message, on_open=self.on_open, on_close=self.on_close, on_error=self.on_error
+        )
+
+    def reconnect(self):
+        while self.should_reconnect:
+            try:
+                self.connect()
+                self.ws.run_forever()
+            except Exception as e:
+                logging.error(f"WebSocket connection failed: {e}")
+
+            logging.info(f"Attempting to reconnect in {self.reconnect_interval} seconds...")
+            time.sleep(self.reconnect_interval)
+
+            self.start_ping_sender()
 
     def run(self):
-        self.ping_sender.start()
-        self.ws.run_forever()
+        self.start_ping_sender()
```

Comment:
Then there's no need to include this here since reconnect also handles the first call